presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractArrayBlock.java

```diff
@@ -111,7 +111,7 @@ public Block copyRegion(int position, int length)
         Block newValues = getRawElementBlock().copyRegion(startValueOffset, endValueOffset - startValueOffset);
 
         int[] newOffsets = compactOffsets(getOffsets(), position + getOffsetBase(), length);
-        boolean[] newValueIsNull = compactArray(getValueIsNull(), position + getOffsetBase(), length);
+        boolean[] newValueIsNull = getValueIsNull() == null ? null : compactArray(getValueIsNull(), position + getOffsetBase(), length);
```

Comment:
since `getValueIsNull()` is abstract, you don't know it's cost and don't know if it's OK to call it twice.

same in `isNull(int position)` & other classes